### PR TITLE
Files upload

### PR DESCRIPTION
**Pull request by PuspaRaaz**
- [x] accept profile image in registration request
- [x] create file-entry table and model
- [x] save uploaded image in database and storage

### CODE REVIEW

File: app/Fileentry.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+     * The attributes that should be hidden for arrays.
+     *
+     * @var array
+     */
+    protected $hidden = [
+        'created_at',
+        'updated_at',
+        'deleted_at'
+    ];
+}
```
Comment (sadhakbj) [inline app/Fileentry.php]: Please add a empty line at the end.

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -2,133 +2,319 @@
...
-use Validator;
-
-use App\User;
-use Helper;
+use View;
 
 class UserController extends Controller
 {
     private $helper;
+    private $response;
```
Comment (sadhakbj) [inline app/Http/Controllers/UserController.php]: Please add doc block for all these properties.

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -2,133 +2,319 @@
...
+     * @return \Illuminate\Http\Response
      */
-    public function logout() {
-        $response = [];
-        // get user with provided token
-        $session = Auth::user();
+    public function register(RegisterRequest $request)
+    {
+        $data = $request->only(['name', 'email', 'username', 'phone', 'password']);
+        $data['password'] = Hash::make($data['password']);
```
Comment (sadhakbj) [inline app/Http/Controllers/UserController.php]: Please fix the indentation

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -2,133 +2,319 @@
...
-     * Store a newly created resource in storage.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @return \Illuminate\Http\Response
-     */
-    public function store(Request $request) {
-        return response()->json([
-            'message' => 'Path not found.'
-        ], 404);
+        // Mail::to($user->email)->send(new UserActivation($user));
```
Comment (sadhakbj) [inline app/Http/Controllers/UserController.php]: Remove this comment.

File: app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -2,133 +2,319 @@
...
      * @return \Illuminate\Http\Response
      */
-    public function destroy($id) {
-        return response()->json([
-            'message' => 'Path not found.'
-        ], 404);
+    public function activation(String $token)
+    {
+
+        // get user with provided token
```
Comment (sadhakbj) [inline app/Http/Controllers/UserController.php]: Your method name and docblock should define what you are doing

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -2,133 +2,319 @@
...
+     * @param \Illuminate\Http\Request      $request
      * @return \Illuminate\Http\Response
      */
-    public function create() {
-        return response()->json([
-            'message' => 'Path not found.'
-        ], 404);
+    public function forgetpassword(Request $request)
+    {
+        // get user with given email
```
Comment (sadhakbj) [inline app/Http/Controllers/UserController.php]: Remove this comment too.

File: app/Http/Controllers/UserController.php [hunk 7]
```diff
@@ -2,133 +2,319 @@
...
+     * @param String                $token
+     * @return \Illuminate\View
      */
-    public function show($id) {
-        return response()->json([
-            'message' => 'Path not found.'
-        ], 404);
+    public function reset_forgot_password(String $token)
+    {
+        // get user with provided token
```
Comment (sadhakbj) [inline app/Http/Controllers/UserController.php]: here too

File: app/Http/Controllers/UserController.php [hunk 8]
```diff
@@ -2,133 +2,319 @@
...
+            return View::make('password_reset')->with([
+                'user' => $user,
+                'message' => $this->helper->statusMessage('0025')
+            ]);
+        }
+
+        $user->forget_token = null;
+        $user->active = true;
+        $user->password = Hash::make($request->password);
+        // insert the new instance of user
```
Comment (sadhakbj) [inline app/Http/Controllers/UserController.php]: What about using the create method to create the user ?

File: app/Http/Controllers/UserController.php [hunk 9]
```diff
@@ -2,133 +2,319 @@
...
+        if (!$session) {
+            $response['code'] = '0052';
+            $response['message'] = $this->helper->statusMessage($response['code']);
+            return response($response);
+        }
+
+        $session->delete();
+        $response['code'] = '0020';
+        $response['message'] = $this->helper->statusMessage($response['code']);
+        return response($response);
```
Comment (sadhakbj) [inline app/Http/Controllers/UserController.php]: There should be empty line before return statement. See `PSR-2` standard.

File: app/Http/Requests/LoginRequest.php [hunk 10]
```diff
@@ -28,7 +28,7 @@ public function rules()
     {
         $device_type = parent::only(['device_type'])['device_type'];
         return [
-            'device_token'  =>  'required',//|unique:login_instances,device_token,NULL,id,device_type,'.$device_type
+            'device_token'  =>  'required',//|unique:logininstances,device_token,NULL,id,device_type,'.$device_type
```
Comment (sadhakbj) [inline app/Http/Requests/LoginRequest.php]: What's this comment ?

File: app/LoginInstance.php [hunk 11]
```diff
@@ -12,6 +12,8 @@ class LoginInstance extends Model
     use SoftDeletes;
     use Authenticatable;
 
+    protected $table = "logininstances";
```
Comment (sadhakbj) [inline app/LoginInstance.php]: DOCBlock

File: app/Http/Controllers/UserController.php (hunk 4 above)
Comment (PuspaRaaz) [inline app/Http/Controllers/UserController.php]: can't remove this comment. just keeping it as this to bypass mailing procedure on registration

File: app/Http/Requests/LoginRequest.php (hunk 10 above)
Comment (PuspaRaaz) [inline app/Http/Requests/LoginRequest.php]: saved for future use

File: app/Http/Controllers/UserController.php (hunk 8 above)
Comment (PuspaRaaz) [inline app/Http/Controllers/UserController.php]: here I am updating user's password, not creating a new user

File: app/Http/Controllers/UserController.php (hunk 5 above)
Comment (PuspaRaaz) [inline app/Http/Controllers/UserController.php]: there it was, already

/**
 * verify user's email and finally activates user
 *
 * @param String $token
 * @return \Illuminate\Http\Response
 */

**Pull request merged**